src/amdinfer/workers/rocal.cpp

```diff
@@ -0,0 +1,439 @@
+    throw invalid_argument(
+      "model file argument missing from model load request");
+  }
+  
+  this->handle_ = rocalCreate(batch_size_, RocalProcessMode::ROCAL_PROCESS_CPU, 0, 1);
+  if (rocalGetStatus(this->handle_) != ROCAL_OK) {
+    AMDINFER_LOG_ERROR(
+      logger, "RocalWorker handle creation failed");
+    // Throwing an exception causes server to delete this worker instance.
+    // Client must try again.
+    throw runtime_error(
+      "could not create rocAL context");
+  }
+  deserialize(this->model_path_);
+}
+
+void RocalWorker::doAcquire(ParameterMap* parameters) { (void)parameters; }
+
+BatchPtr RocalWorker::doRun(Batch* batch, [[maybe_unused]]const MemoryPool* pool) {
+#ifdef AMDINFER_ENABLE_LOGGING
+  const auto& logger = this->getLogger();
+#endif
+
+  if (rocalGetStatus(this->handle_) != ROCAL_OK) {
+      std::cerr << "Rocal graph not properly intialized" << std::endl;
+      return nullptr;
+  }
+  util::Timer timer;
+  timer.add("rocal_batch_start");
+
+  std::vector<unsigned char *> input_batch_buffer; 
+
+  for (unsigned int j = 0; j < batch->size(); j++) {
+      const auto& req = batch->getRequest(j);
+      const auto& inputs = req->getInputs();
+  
+      if (inputs.size() != 1) {
+          req->runCallbackError("Only one input tensor should be present");
+          continue;
+      }
+      const auto& input = inputs.at(0);
+      const auto& input_shape = input.getShape();
+
+      assert(input_shape.size() == 1); // compressed raw buffer should have shape length = 1
+      ROI_xywh_[j].h = input_shape[0];
+      auto input_data = static_cast<unsigned char*>(input.getData());
+      input_batch_buffer.push_back(input_data);
+  }
+
+  bool eos = true; // eos value is always true since there will be only one input tensor 
+  rocalExternalSourceFeedInput(handle_, {}, false, input_batch_buffer, ROI_xywh_, max_decode_width_, max_decode_height_, channels_, RocalExternalSourceMode(rocal_external_mode_), 
+                                RocalTensorLayout(0), eos);
+
+  AMDINFER_LOG_INFO(logger, "Beginning rocalRun eval");
+
+  timer.add("eval_start");
+  if (rocalRun(this->handle_) != ROCAL_OK) {
+      AMDINFER_LOG_ERROR(logger, "rocalRun failed to run");
+      throw runtime_error("rocalRun failed to run");
+  }
+  timer.add("eval_end");
+
+  auto eval_duration_us = timer.count<std::micro>("eval_start", "eval_end");
+  [[maybe_unused]] auto eval_duration_s = eval_duration_us / std::mega::num;
+  AMDINFER_LOG_INFO(
+    logger,
+    std::string("Finished rocalRun eval; batch size: ") +
+      std::to_string(batch_size_) +
+      "  elapsed time: " + std::to_string(eval_duration_us) +
+      " us.  Images/sec: " + std::to_string(batch_size_ / (eval_duration_s)));
+
+
+  BatchPtr new_batch = batch->propagate();
+  
+  RocalTensorList output_tensor_list = rocalGetOutputTensors(handle_);
+  assert(output_tensor_list->size() == batch->size());
+
+  // RocalTensor output_tensor = output_tensor_list->at(0);
```

Comment:
please remove these commented lines if not used

Reply:
Removed